.github/workflows/main.yml

```diff
@@ -0,0 +1,26 @@
+name: King of Tokyo
+
+on:
+  push:
+    branches: [ 'main' ]
+
+jobs:
+  build-image:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+      # Build docker-compose image
+      - name: Build docker-compose image
+        run: docker-compose build web
+      # Log in to Docker Hub
+      - name: Login to Docker Hub
+        env: 
+          DOCKER_USERNAME: ${{secrets.DOCKER_USERNAME}}
+          DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+        run: docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+      # Tag the Docker image
+      - name: Tag the Docker image
+        run: docker tag king-of-tokyo-webapi-team2_web:latest ${{secrets.DOCKER_USERNAME}}/kot-image:$GITHUB_RUN_ID
+      # Push the Docker image to Docker Hub
+      - name: Push the Docker image
+        run: docker push ${{secrets.DOCKER_USERNAME}}/kot-image:$GITHUB_RUN_ID
```

Comment:
Here you need to make sure to remove the new image built and pushed and then make sure to log out your docker session.